sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala

```diff
@@ -36,6 +37,7 @@ class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
    * @since 2.0.0
    */
   def set(key: String, value: String): Unit = {
+    assertNotGlobalSQLConf(key)
```

Comment:
all global sql conf will be automatically registered, we don't need to change the code here when adding more global sql conf